spec/system/composer/default_to_subcategory_spec.rb

```diff
@@ -6,32 +6,66 @@
   fab!(:user) { Fabricate(:user) }
   fab!(:group) { Fabricate(:group) }
   fab!(:group_user) { Fabricate(:group_user, user: user, group: group) }
+  fab!(:default_latest_category) { Fabricate(:category) }
   fab!(:category) { Fabricate(:private_category, group: group, permission_type: 3) }
   fab!(:subcategory) do
     Fabricate(:private_category, parent_category_id: category.id, group: group, permission_type: 1)
   end
+  fab!(:category_with_no_subcategory) do
+    category_with_no_subcategory = Fabricate(:private_category, group: group, permission_type: 3)
+    category_with_no_subcategory.read_restricted = false
+    category_with_no_subcategory.save
+    category_with_no_subcategory
+  end
   let(:category_page) { PageObjects::Pages::Category.new }
-  before { sign_in(user) }
-
-  describe "Setting enabled and can't post on parent category" do
-    before { SiteSetting.default_subcategory_on_read_only_category = true }
 
-    it "should have 'New Topic' button enabled and default Subcategory set in the composer" do
-      category_page.visit(category)
-      expect(category_page).to have_button("New Topic", disabled: false)
-      category_page.new_topic_button.click
-      select_kit =
-        PageObjects::Components::SelectKit.new(page.find("#reply-control.open .category-chooser"))
-      expect(select_kit).to have_selected_value(subcategory.id)
+  describe "anon user" do
+    it "can visit the category" do
+      category_page.visit(category_with_no_subcategory)
+      select_kit = PageObjects::Components::SelectKit.new(page.find(".navigation-container"))
+      expect(select_kit).to have_selected_value(category_with_no_subcategory.id)
     end
   end
 
-  describe "Setting disabled and can't post on parent category" do
-    before { SiteSetting.default_subcategory_on_read_only_category = false }
+  describe "logged in user" do
+    before { sign_in(user) }
+    describe "Setting enabled and can't post on parent category" do
```

Comment:
I think this description can be improved further to indicate the exact site settings that are enabled.